elastic/tests/test_elastic.py

```diff
@@ -220,6 +215,30 @@ def test_aws_auth(instance, expected_aws_host, expected_aws_service):
     assert 'aws_host' not in ESCheck.HTTP_CONFIG_REMAPPER
 
 
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    'instance, expected_aws_host, expected_aws_service',
+    [
+        pytest.param({}, None, None, id='not aws auth'),
+        pytest.param(
+            {'auth_type': 'aws', 'aws_region': 'foo', 'aws_host': 'foo.com'},
+            'foo.com',
+            'es',
+            id='aws_host_custom_no_url',
+        ),
+    ],
+)
+def test_aws_auth_no_url(instance, expected_aws_host, expected_aws_service):
+    with pytest.raises(ConfigurationError):
+        check = ESCheck('elastic', {}, instances=[instance])
+
+        assert getattr(check.http.options.get('auth'), 'aws_host', None) == expected_aws_host
+        assert getattr(check.http.options.get('auth'), 'service', None) == expected_aws_service
+
+        # make sure class attribute HTTP_CONFIG_REMAPPER is not modified
+        assert 'aws_host' not in ESCheck.HTTP_CONFIG_REMAPPER
```

Comment:
Do these assertions actually get run? I'd expect not because `ConfigurationError` is triggered by instantiating `ESCheck`.

I'm actually not sure these assertions are useful, since if `ConfigurationError` is raised then the check simply won't run and we don't care how HTTP auth was configured, right?

```suggestion
```

Reply:
Make sense, I'll remove the `test_aws_auth_no_url`